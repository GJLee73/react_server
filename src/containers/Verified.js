/**
 * 인증되었다는 것을 보여주는 페이지. 서버의 verified와 연결되어 있음
 * 
 * @author G1
 * @logs // 18.2.25
 */
import React from 'react';
import {Link} from 'react-router-dom'

const Verified = () => {
    return (
        <div>
            인증되었습니다! 다시 로그인해주세요 
            <p><Link to ='/login'>로그인하기</Link></p>
        </div>
    )
}
 
export default Verified;