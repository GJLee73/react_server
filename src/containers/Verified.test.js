import React from 'react';
import {Verified} from './Verified'//test를 위해서는 reux와의 연결을 해지해주어야 한다.
import { shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom'

jest.mock('../mocks')


describe('<Verified>', () => {
  const verify = jest.fn(()=>Promise.resolve())
  const logoutRequest = jest.fn()
  var wrapper =shallow(<Verified verify={verify} match={{params:{token:'1234'}}} currentUser={'g1'}
                                  logoutRequest={logoutRequest}/>);

    describe('function test', () => {
      it('componetWillMount test',()=>{
        expect(verify).toHaveBeenCalled()
        expect(logoutRequest).toHaveBeenCalled()
      })
    });
    it('snapshot test', () => {
      expect(wrapper).toMatchSnapshot();
    })
  
  });