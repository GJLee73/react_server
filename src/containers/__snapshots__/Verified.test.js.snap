// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Verified> snapshot test 1`] = `
<div>
  인증되었습니다! 다시 로그인해주세요
  <p>
    <Link
      replace={false}
      to="/login"
    >
      로그인하기
    </Link>
  </p>
</div>
`;
